software/motion.py

```diff
@@ -0,0 +1,64 @@
+import math
+import numpy as np
+import tkinter as tk
+import serial
+import struct
+import serial.tools.list_ports
+import time as time
+
+class IRobotMotion:
+    def open(self):
+        pass
+    def close(self):
+        pass
+    def move(self, x_speed, y_speed, rot_speed):
+        pass
+
+class OmniMotionRobot(IRobotMotion):
+
+    def __init__(self):
+        pid = 22336
+        for x in serial.tools.list_ports.comports():
+            if x.pid == pid:
+                self.device = x.device
+
+        self.serialObj = serial.Serial()
+
+        self.wheel_distance_from_center = 0.2
+        self.wheelSpeedToMainboardUnits = 90.9
+
+    def open(self):
+        self.serialObj.port = self.device
+        self.serialObj.open()
+
+    def move(self, x_speed, y_speed, rot_speed):#////////////////////////
+        robotSpeed = math.sqrt(x_speed * x_speed + y_speed * y_speed)
+        robotDirectionAngle = math.atan2(y_speed, x_speed)
+
+        wheelLinearVelocities = []
+        for angle in [0,120,240]:
+            wheelLinearVelocities.append(robotSpeed * math.cos(robotDirectionAngle - math.radians(angle)) + self.wheel_distance_from_center * rot_speed)
+        
+        wheelAngularSpeedMainboardUnits = []
+        for wheelLinearVelocity in wheelLinearVelocities:
+            wheelLinearVelocity = wheelLinearVelocity * self.wheelSpeedToMainboardUnits
+            if -1 < wheelLinearVelocity < 0:
```

Comment:
Not sure why this if-else is required when you round it using the int cast anyways